src/main/scala/mesosphere/marathon/core/storage/store/impl/cache/LazyCachingPersistenceStore.scala

```diff
@@ -159,3 +162,191 @@ class LazyCachingPersistenceStore[K, Category, Serialized](
 
   override def toString: String = s"LazyCachingPersistenceStore($store)"
 }
+
+case class LazyVersionCachingPersistentStore[K, Category, Serialized](
+    store: PersistenceStore[K, Category, Serialized],
+    config: PersistenceStorageConfig.VersionCacheConfig = PersistenceStorageConfig.VersionCacheConfig.Default)(implicit
```

Comment:
mostly a nit, but I generally prefer passing the named values in at this level (instead of passing the config) - doesn't really matter.